pkg/ruler/ruler.go

```diff
@@ -507,8 +536,24 @@ func instanceOwnsRuleGroup(r ring.ReadRing, g *rulespb.RuleGroupDesc, disabledRu
 			}
 		}
 	} else {
-		// Even if the replication factor is set to a number bigger than 1, only the first ruler evaluates the rule group
 		ownsRuleGroup = rlrs.Instances[0].Addr == instanceAddr
+		if r.Config().EnableHA {
+			for i, ruler := range rlrs.Instances {
+				if ruler.Addr == instanceAddr && i == 0 {
+					level.Debug(r.Logger()).Log("msg", "primary taking ownership", "user", g.User, "group", g.Name, "namespace", g.Namespace, "token", hash, "ruler", instanceAddr)
```

Comment:
I wonder if it really helps to log `token` here. Does it help debug HA issues?

Reply:
It helps issues with troubleshooting evaluation issues. Users can filter logs by token and know what happened with a particular rule group

Reply:
> Users can filter logs by token and know what happened with a particular rule group

Are we going to document this clearly? Otherwise I don't think it is something easy to debug as a user. Tbh I don't how I map a particular rule group to a hash by just looking at the log.
Isn't `group` and `namespace` already enough to locate a particular rule group?

Reply:
Removed